lib/improver/tests/nowcasting/lightning/test_NowcastLightning.py

```diff
@@ -0,0 +1,946 @@
+        self.precip_cube.data[1:, 0, ...] = 0.
+        # iris.util.queeze is applied here to demote the singular coord "time"
+        # to a scalar coord.
+        self.vii_cube = squeeze(
+            add_forecast_reference_time_and_forecast_period(
+                set_up_cube(num_realization_points=3,
+                            zero_point_indices=[]),
+                fp_point=0.0))
+        threshold_coord = self.vii_cube.coord('realization')
+        threshold_coord.points = [0.5, 1.0, 2.0]
+        threshold_coord.rename('threshold')
+        threshold_coord.units = cf_units.Unit('kg m^-2')
+        self.vii_cube.data = np.zeros_like(self.vii_cube.data)
+
+    def test_basic(self):
+        """Test that the method returns the expected cube type"""
+        plugin = Plugin()
+        result = plugin._modify_first_guess(self.cube,
+                                            self.fg_cube,
+                                            self.ltng_cube,
+                                            self.precip_cube,
+                                            None)
+        self.assertIsInstance(result, Cube)
+
+    def test_basic_with_vii(self):
+        """Test that the method returns the expected cube type"""
+        plugin = Plugin()
+        result = plugin._modify_first_guess(self.cube,
+                                            self.fg_cube,
+                                            self.ltng_cube,
+                                            self.precip_cube,
+                                            self.vii_cube)
+        self.assertIsInstance(result, Cube)
+
+    def test_input(self):
+        """Test that the method does not modify the input cubes."""
+        plugin = Plugin()
+        cube_a = self.cube.copy()
+        cube_b = self.fg_cube.copy()
+        cube_c = self.ltng_cube.copy()
+        cube_d = self.precip_cube.copy()
+        plugin._modify_first_guess(cube_a, cube_b, cube_c, cube_d, None)
+        self.assertArrayAlmostEqual(cube_a.data, self.cube.data)
+        self.assertArrayAlmostEqual(cube_b.data, self.fg_cube.data)
+        self.assertArrayAlmostEqual(cube_c.data, self.ltng_cube.data)
+        self.assertArrayAlmostEqual(cube_d.data, self.precip_cube.data)
+
+    def test_input_with_vii(self):
+        """Test that the method does not modify the input cubes."""
+        plugin = Plugin()
+        cube_a = self.cube.copy()
+        cube_b = self.fg_cube.copy()
+        cube_c = self.ltng_cube.copy()
+        cube_d = self.precip_cube.copy()
+        cube_e = self.vii_cube.copy()
+        plugin._modify_first_guess(cube_a, cube_b, cube_c, cube_d, cube_e)
+        self.assertArrayAlmostEqual(cube_a.data, self.cube.data)
+        self.assertArrayAlmostEqual(cube_b.data, self.fg_cube.data)
+        self.assertArrayAlmostEqual(cube_c.data, self.ltng_cube.data)
+        self.assertArrayAlmostEqual(cube_d.data, self.precip_cube.data)
+        self.assertArrayAlmostEqual(cube_e.data, self.vii_cube.data)
```

Comment:
You can probably get away with losing `test_input`, since this covers the same and also includes the conditional vii stuff.

Reply:
Yes, I think you're right. I'll delete that one.